tests/IResearch/AgencyMock.cpp

```diff
@@ -22,6 +22,7 @@
 /// @author Vasiliy Nabatchikov
 ////////////////////////////////////////////////////////////////////////////////
 
+#define ENABLE_MOCKS
```

Comment:
Please make the name more specific and add a comment what this is for.

Reply:
The original thought was that there could be more instances where code would be made public to enable mocking, so that it could apply elsewhere. On second thought, making it specific instead, poses less burden on the reader. Will incorporate your suggestion and make it MAKE_NOTIFY_OBSERVERS_PUBLIC.